cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -1527,4 +1531,30 @@ private void emitTimeMetrics(String namespace, String appName, String metricName
     long timeTaken = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTime);
     metricsCollectionService.getContext(tags).gauge(metricName, timeTaken);
   }
+
+  /**
+   * Update scm metadata for the given apps.
+   * @param namespaceId namespace where the apps are deployed
+   * @param updateRequests list of {@link AppUpdateGitMetaRequest}
+   * @throws IOException when the update operation fails
+   */
+  public void updateGitMetaMulti(NamespaceId namespaceId, List<AppUpdateGitMetaRequest> updateRequests)
+      throws IOException {
+    List<UpdateSourceControlMetaRequest> updateScmMetaRequests = new ArrayList<>();
+    for (AppUpdateGitMetaRequest updateRequest : updateRequests) {
+      ApplicationId appId = new ApplicationId(
```

Comment:
It is possible for this to throw an exception for invalid application ids. Need to catch it and throw BadRequestException so that it becomes a 400.

It is better to do this type of user input validation/conversion in the http handler, and have the service layer (and below) only deal with valid objects.

Reply:
done.